app/models/service.rb

```diff
@@ -339,6 +339,7 @@ def reconfigure_resource_action
   end
 
   def reconfigure_dialog
+    raise unsupported_reason(:reconfigure) unless supports_reconfigure?
```

Comment:
Interesting.  I like this.  What is `unsupported_reason(:reconfigure)`?  Could you add a test if there isn't one?  I'd like to see how the api change will look using this.